app/models/legacy/project_event.rb

```diff
@@ -53,7 +57,17 @@ def migration_step_data
     def migrate_step
       data = migration_step_data
       puts "ProjectStep[#{data[:id]}] #{data[:project_id]}"
-      ::ProjectStep.create(data)
+      step = ::ProjectStep.create(data)
+      step.parent = find_or_create_parent_group
+      step.save!
+      step
+    rescue StandardError => e
+      $stderr.puts "#{step.class.name}[#{id}] error migrating step, could not find loan #{data[:project_id]}: #{e} - skipping"
+    end
+
+    def find_or_create_parent_group
+      return unless loan
+      ProjectGroup.find_or_create_by(project_type: "Loan", project_id: loan.id)
```

Comment:
I added the suggested comment on line 61. That is also the only place this method is called. So at that point `parent_id` will alway be nil. Later in the script, we will assign a different `parent_id` if needed. Does this warrant another comment here?